drivers/include/net/netdev/layered.h

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2017 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_netdev_api
+ * @{
+ *
+ * @file
+ * @brief       Netdev layer helper functions.
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ */
+
+#ifndef NET_NETDEV_H
+#define NET_NETDEV_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "net/netdev.h"
+#include "assert.h"
+
+/**
+ * @brief   Add a netdev layer to the netdev layer stack
+ *
+ * @param[in] top     Last netdev device added to the stack.
+ * @param[in] dev     New layer to push to the top of the stack.
+ *
+ *
+ * @return  The new top netdev layer of the netdev stack.
+ */
+static inline netdev_t *netdev_add_layer(netdev_t *top, netdev_t *netdev)
+{
+    assert(top != NULL);
+    netdev->lower = top;
+    top->context = netdev;
+    return netdev;
+}
+
+/**
+ * @brief   Passthrough init function.
+ *
+ * See also @ref netdev_driver for the extended description of this functions
+ * behaviour
+ *
+ * @param[in]   dev     network device descriptor
+ * @return              `< 0` on error, 0 on success
+ */
+int netdev_init_pass(netdev_t *dev)
+{
```

Comment:
You should still move those into a C-file ;-).

Reply:
Let's try that again...